django_mongodb_backend/cache.py

```diff
@@ -1,27 +1,33 @@
 import pickle
 from datetime import datetime, timezone
+from base64 import b64encode, b64decode
 
 from django.core.cache.backends.base import DEFAULT_TIMEOUT, BaseCache
 from django.core.cache.backends.db import Options
+from django.core.signing import Signer, BadSignature
 from django.db import connections, router
 from django.utils.functional import cached_property
 from pymongo import ASCENDING, DESCENDING, IndexModel, ReturnDocument
 from pymongo.errors import DuplicateKeyError, OperationFailure
 
 
 class MongoSerializer:
-    def __init__(self, protocol=None):
+    def __init__(self, protocol=None, signer=True, salt=None):
         self.protocol = pickle.HIGHEST_PROTOCOL if protocol is None else protocol
+        self.signer = Signer(salt=salt) if signer else None
 
     def dumps(self, obj):
         # For better incr() and decr() atomicity, don't pickle integers.
         # Using type() rather than isinstance() matches only integers and not
         # subclasses like bool.
         if type(obj) is int:  # noqa: E721
-            return obj
-        return pickle.dumps(obj, self.protocol)
+            return obj if self.signer is None else self.signer.sign(b64encode(obj))
+        pickled_data = pickle.dumps(obj, protocol=self.protocol)  # noqa: S301
+        return self.signer.sign(b64encode(pickled_data).decode()) if self.signer else pickled_data
 
     def loads(self, data):
+        if self.signer is not None:
+            data = b64decode(self.signer.unsign(data))
```

Comment:
```suggestion
        if self.signer is not None:
            if type(unsigned_data := self.signer.unsign(data)) is int:
                data = unsigned_data
            else:
                data = b64decode(unsigned_data)
```

Reply:
I considered this when I was originally writing the function, but `1234` is valid base64 and `int`. So, it would be ambiguous as to which was actually being stored. I fixed the issue by simply not signing `int` values. I do not believe this to introduce any real problems and solves the `$inc` test cases which were failing.